services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -124,11 +124,30 @@ function checkUnicity(primaryKeys, uniqueIndexes, columnName) {
       indexColumnName.length === 1 && indexColumnName.includes(columnName));
 
   const isPrimary = _.isEqual([columnName], primaryKeys);
-  return { isPrimary, isUnique };
+
+  return isPrimary || isUnique;
+}
+
+function associationNameAlreadyExists(existingReferences, newReference) {
+  return existingReferences.some((reference) => reference.as === newReference.as);
+}
+
+function referenceAlreadyExists(existingReferences, newReference) {
+  return existingReferences.some((reference) => (
+    reference.ref === newReference.ref
```

Comment:
I got a `null` value for `reference` while testing. Not sure of the reason, but worth investigating.